lib/backpex/fields/upload.ex

```diff
@@ -1,124 +1,163 @@
 defmodule Backpex.Fields.Upload do
-  @moduledoc ~S"""
-  A field for handling uploads.
-
-  > #### Warning {: .warning}
-  >
-  > This field does **not** currently support `Phoenix.LiveView.UploadWriter` and direct / external uploads.
-
-  ## Options
-
-    * `:upload_key` (atom) - Required identifier for the upload field (the name of the upload).
-    * `:accept` (list) - Required filetypes that will be accepted.
-    * `:max_entries` (integer) - Required number of max files that can be uploaded.
-    * `:max_file_size` (integer) - Optional maximum file size in bytes to be allowed to uploaded. Defaults 8 MB (`8_000_000`).
-    * `:list_existing_files` (function) - Required function that returns a list of all uploaded files based on an item.
-    * `:file_label` (function) - Optional function to get the label of a single file.
-    * `:consume_upload` (function) - Required function to consume file uploads.
-    * `:put_upload_change` (function) - Required function to add file paths to the params.
-    * `:remove_uploads` (function) - Required function that is being called after saving an item to be able to delete removed files
-
-
-  > #### Info {: .info}
-  >
-  > The following examples copy uploads to a static folder in the application. In a production environment, you should consider uploading files to an appropriate object store.
-
-  ## Options in detail
-
-  The `upload_key`, `accept`, `max_entries` and `max_file_size` options are forwarded to https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#allow_upload/3. See the documentation for more information.
-
-  ### `list_existing_files`
-
-  **Parameters**
-  * `:socket` - The socket.
-  * `:item` (struct) - The item without its changes.
-
-  The function is being used to display existing uploads. The function receives the socket and the item and has to return a list of strings. Removed files during an edit of an item are automatically removed from the list. This option is required.
-
-  **Example**
-
-      def list_existing_files(_socket, item), do: item.files
-
-  ### `file_label`
-
-  **Parameters**
-  * `:file` (string) - The file.
-
-  The function can be used to modify a file label based on a file. In the following example each file will have an "_upload" suffix. This option is optional.
-
-  **Example**
-
-      def file_label(file), do: file <> "_upload"
-
-  ### `consume_upload`
-
-  **Parameters**
-  * `:socket` - The socket.
-  * `:item` (struct) - The saved item (with its changes).
-  * `:meta` - The upload meta.
-  * `:entry` - The upload entry.
+  @config_schema [
+    upload_key: [
+      doc: "Required identifier for the upload field (the name of the upload).",
+      type: :atom
+    ],
+    accept: [
+      doc: "Required filetypes that will be accepted.",
+      type: {:list, :string}
```

Comment:
`accept` can also be of type `:atom` (= `:any`). See https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#allow_upload/3-options